### PR TITLE
LX-77 Add support for JIT provisioning pt 1

### PR DESCRIPTION
**Pull request by jeskepetr**

### CODE REVIEW

File: gooddata-server-oauth2-autoconfigure/src/main/kotlin/AuthenticationUtils.kt [hunk 1]
```diff
@@ -233,7 +233,8 @@ private fun ClientRegistration.resolveSupportedScopes() =
 
 private fun ClientRegistration.Builder.withScopes(supportedScopes: Scope?): ClientRegistration.Builder {
     // in the future, we could check mandatory scopes against the supported ones
-    val mandatoryScopes = listOf(OIDCScopeValue.OPENID, OIDCScopeValue.PROFILE).map(Scope.Value::getValue)
+    val mandatoryScopes =
+        listOf(OIDCScopeValue.OPENID, OIDCScopeValue.PROFILE, OIDCScopeValue.EMAIL).map(Scope.Value::getValue)
```
Comment (mverbovsky) [inline gooddata-server-oauth2-autoconfigure/src/main/kotlin/AuthenticationUtils.kt:237]: Can not be the email optional?

File: gooddata-server-oauth2-autoconfigure/src/main/kotlin/OktaRegistrationProperties.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package com.gooddata.oauth2.server
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.boot.context.properties.bind.DefaultValue
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "spring.security.oauth2.client.registration.okta")
+class OktaRegistrationProperties(
```
Comment (mverbovsky) [inline gooddata-server-oauth2-autoconfigure/src/main/kotlin/OktaRegistrationProperties.kt:24]: Why do you need it?

File: gooddata-server-oauth2-autoconfigure/src/main/kotlin/JitProvisioningAuthenticationSuccessHandler.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        authenticationToken: OAuth2AuthenticationToken,
+        webFilterExchange: WebFilterExchange?,
+    ): Mono<*> {
+        return mono {
+            val organization = authenticationStoreClient.getOrganizationByHostname(
+                webFilterExchange?.exchange?.request?.uri?.host ?: ""
+            )
+            if (organization.jitEnabled == true) {
+                checkMandatoryClaims(authenticationToken)
+                logger.info { "action=JIT message=Initiating JIT provisioning." }
```
Comment (mverbovsky) [inline gooddata-server-oauth2-autoconfigure/src/main/kotlin/JitProvisioningAuthenticationSuccessHandler.kt:54]: You should use this type of logging
```
            logger.logInfo {
                withMessage { "User attempts to authenticate" }
                withAction("login")
                withState("started")
                withOrganizationId(organization.id)
            }
```

File: gooddata-server-oauth2-autoconfigure/src/main/kotlin/OktaRegistrationProperties.kt (hunk 2 above)
Comment (jeskepetr) [reply gooddata-server-oauth2-autoconfigure/src/main/kotlin/OktaRegistrationProperties.kt]: Good catch, this was some left over from my initial experiments, removed!

File: gooddata-server-oauth2-autoconfigure/src/main/kotlin/AuthenticationUtils.kt (hunk 1 above)
Comment (jeskepetr) [reply gooddata-server-oauth2-autoconfigure/src/main/kotlin/AuthenticationUtils.kt]: Makes sense. I've added `gd_user_groups` scope and configured both `email` & `gd_users_groups `scopes as optional.

**Review by mverbovsky — APPROVED** [review]

**Pull request merged**